presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -155,13 +155,19 @@ private boolean hasPreviousSpillCompletedSuccessfully()
     public ListenableFuture<?> startMemoryRevoke()
     {
         checkState(spillInProgress.isDone());
+        if (hashAggregationBuilder.hasBuiltFinalResult()) {
+            return spillInProgress;
+        }
         spillToDisk();
         return spillInProgress;
     }
 
     @Override
     public void finishMemoryRevoke()
     {
+        if (hashAggregationBuilder.hasBuiltFinalResult()) {
+            return;
+        }
```

Comment:
Is this part necessary?

Reply:
Guess not... not sure what I was thinking here anymore 😆

Reply:
Yeah since memory usage is updated in `updateMemory()` to be set to the size of the empty hash agg builder. This is not correct since we never spilt in the first place in `startMemoryRevoke`.

Reply:
Adding comments.